src/App.tsx

```diff
@@ -1,15 +1,35 @@
 import { Route, Routes } from 'react-router';
 import { useLoadingContext } from './contexts/LoadingContext';
+import { useAuthContext } from './contexts/AuthContext';
+import axios from 'axios';
 
 import SignIn from './pages/SignIn';
 import SignUp from './pages/SignUp';
 import Vote from './pages/Vote';
 import Result from './pages/Result';
 
 import { Spinner } from './components/Spinner';
+import { useEffect } from 'react';
 
 function App() {
   const { loading }: any = useLoadingContext();
+  const { setIsAuth, setUserData }: any = useAuthContext();
+
+  // initialize fucntion
+  const initializeUserInfo = () => {
+    const ud = JSON.stringify(localStorage.getItem('userData'));
+    const token = JSON.stringify(localStorage.getItem('token'));
+    if (!ud) return;
+    setUserData(ud);
+    setIsAuth(true);
+    axios.defaults.headers.common['Authorization'] = token;

```

Comment:
```suggestion
    if (ud) {
    setUserData(ud);
    setIsAuth(true);
    axios.defaults.headers.common['Authorization'] = token;
    }
```

이렇게 쓰는 게 가독성이 더 좋을 것 같습니당

```suggestion
    (ud) && { 
    setUserData(ud);
    setIsAuth(true);
    axios.defaults.headers.common['Authorization'] = token;
    }
```

아니면 이렇게요~

Reply:
good